metafields_test.go

```diff
@@ -0,0 +1,103 @@
+package shopify
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/jaswdr/faker"
+)
+
+/**
+GIVEN I have a single metafield
+WHEN I ask to get that metafield by its key when its of type boolean
+THEN the value returned is of the correct type and what is expected
+*/
+func TestMetafields_GetByKeyBoolean(t *testing.T) {
```

Comment:
You could define a `testCases` structure like below (I just done this very quickly, I would need to look into each case to define one that can handle all cases). You can then loop over the slice and run tests in its own routine. You might not even need to use faker for this. I think it is okay to just hard code the metafields data.

Based on the `Type` or some other field you can then go into different if statements that can check the expected value against the got value. Laying out tests like this is quite common in Go source code.

```go
var testCases []struct {
	Metafield      struct{}
	Type           string
	ExpectedInt    int
	ExpectedBool   bool
	ExpectedString string
}{
	{},
	{},
}

// For loop
```

Reply:
That's interesting. As long as you can run a test solely without it being one test running all of them. I might leave it now though because there are only a few.

Reply:
So you would pass `t *testing.T` and then in the for loop call `t.Run()`. I think if more test are going to be added in the future, its easier to understand in the above structure then loads of test functions.

```
func (*testing.T).Run(name string, f func(t *testing.T)) bool
Run runs f as a subtest of t called name. It runs f in a separate goroutine and blocks until f returns or calls t.Parallel to become a parallel test. Run reports whether f succeeded (or at least did not fail before calling t.Parallel).

Run may be called simultaneously from multiple goroutines, but all such calls must return before the outer test function for t returns.
```

Reply:
I will remember this.